### PR TITLE
Refactor sidebar

### PR DESCRIPTION
**Pull request by abocquet**
Stop using `use` for the sidebar and drop interfaces to use JSX elements instead. Makes it more flexible and simpler.

### CODE REVIEW

File: frontend/src/components/associations/Router.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import { api, useBetterQuery } from "../../services/apiService";
+import { PrivateRoute } from "../utils/Route";
+import { routes } from "../../routing/associations";
+import { ErrorMessage, PageNotFoundError } from "../utils/ErrorPage";
+import { Association } from "../../models/associations/association";
+import { AssociationLayout } from "./Layout";
+import { Loading } from "../utils/Loading";
+
+export const AssociationRouter = ({ match }) => {
+    let { associationId } = useParams<{ associationId: string }>();
```
Comment (Piwakk) [inline frontend/src/components/associations/Router.tsx:18]: const

File: frontend/src/components/associations/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import { AssociationSidebar } from "./Sidebar";
+import Container from "react-bootstrap/Container";
+import React from "react";
+import { Association } from "../../models/associations/association";
+
+export const AssociationLayout = ({
```
Comment (Piwakk) [inline frontend/src/components/associations/Layout.tsx:8]: Maybe add a comment here to explain how to use this?

Comment (Piwakk) [reply frontend/src/components/associations/Layout.tsx:8]: For instance, I would personally like to know what kind of `additionalSidebar` should be passed on (with a separator, in a container, in a row, etc.)

File: frontend/src/components/utils/sidebar/CheckboxField.tsx [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
             className="custom-control-input"
             type="checkbox"
-            checked={state}
-            onChange={() => setState(!state)}
+            checked={state[id]}
+            onChange={() => {
+                let newState = { ...state };
+                newState[id] = !state[id];
+                setState(newState);
+            }}
```
Comment (Piwakk) [inline frontend/src/components/utils/sidebar/CheckboxField.tsx:14]:
```suggestion
            onChange={() => setState({
                ...state,
                [id]: !state[id]
            })};
```
or something like this.
https://stackoverflow.com/a/2274327

File: frontend/src/components/utils/sidebar/SidebarSection.tsx [hunk 4]
```diff
@@ -20,13 +15,9 @@ export const SidebarSection = ({
          * the section should always be visible independently from the default,
          * hence the "&& !retractable"
```
Comment (Piwakk) [inline frontend/src/components/utils/sidebar/SidebarSection.tsx:16]: The comment should be changed?

File: frontend/src/routing/associations.ts [hunk 5]
```diff
@@ -14,85 +14,109 @@ import { AssociationMarketplaceProductAdministration } from "../components/assoc
...
+ * which is `defaultLayout`. If `defaultLayout` is `true`, the `component` will
+ * be automatically be wrapped in an `AssociationLayout`, which includes a sidebar
+ * by default. If it's `false` it will not.
+ * You may want to take advantage of it if :
+ * - you want a custom sidebar, in that case you'll wrap your component with
+ * `AssociationLayout` manually
+ * - you don't want a sidebar at all, in that case you can organise your code as
+ * usual
+ */
 export const routes = (association) => [
```
Comment (Piwakk) [inline frontend/src/routing/associations.ts:28]: Would it be possible to type `routes`?
I guess something like
`(association: Association) => {path: string; component: any; exact: boolean; props: {[key:string]: any}; defaultLayout?: boolean}`

File: frontend/src/components/associations/Router.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    // Generate the routes
+    const privateRoutes = routes(association).map(
+        ({
+            path,
+            component: Component,
+            exact,
+            props,
+            defaultLayout = false,
+        }) => {
+            let renderedComponent = defaultLayout
```
Comment (Piwakk) [inline frontend/src/components/associations/Router.tsx:34]: const

File: frontend/src/components/utils/sidebar/CheckboxField.tsx (hunk 3 above)
Comment (abocquet) [reply frontend/src/components/utils/sidebar/CheckboxField.tsx:14]: Niceeee

**Pull request merged**